spring/boot2-autoconfigure/src/test/resources/config/application-disabled.yml

```diff
@@ -0,0 +1,2 @@
+armeria:
+  server-enabled: false
```

Comment:
hm. I'm not sure if we ever need this property.
The application has this dependency to use an Armeria server. If they don't need an Armeria server, then they do not have to include this dependency. Isn't that right? 🤔

Reply:
In our team we provide some client side autoconfigure for grpc stub and webclient, some thing like https://github.com/line/armeria/pull/3280. But some projects only need the client features(only need grpc clients).
Another option is to seperate the client side  and server side autoconfigure in different module.  #3280 seems to put them together.

Reply:
Thanks for the explanation.
Yeah, if Spring autoconfigure has client-specific settings such as #3280, we might need this `server-enabled` property unless we split the code.

The client-specific settings do not have much code and Spring autoconfigure module does not have many dependencies, so it's probably not a good idea to split the client module.

So I guess it all depends on whether #3280 is merged or not. 🤔